### PR TITLE
Save settings to a txt file

### LINKED ISSUE #204: Save user settings in a txt file

### PR DESCRIPTION
**Pull request by adhy-p**
closes #204

### CODE REVIEW

File: src/main/java/storage/DataFileReader.java [hunk 1]
```diff
@@ -200,7 +207,11 @@ private void extractFromDirectory(Path directoryPath) throws IOException {
             }
 
             try {
-                extractCheatSheet(dataDirectoryFile);
+                if (dataDirectoryFile.getName().equals("settings.txt")) {
+                    loadUserSettings(dataDirectoryFile);
+                } else {
+                    extractCheatSheet(dataDirectoryFile);
+                }
```
Comment (theopin) [inline src/main/java/storage/DataFileReader.java:214]: Good distinction of the settings file from the others

File: src/main/java/storage/DataFileReader.java [hunk 2]
```diff
@@ -335,4 +346,29 @@ private void createNewCheatSheet(boolean isMarkedFavourite,
         newCheatSheet.setFavourite(isMarkedFavourite);
         cheatSheetList.add(newCheatSheet);
     }
+
+    /**
+     * Reads the save file line by line and passes each line to applySettings method.
+     */
```
Comment (theopin) [inline src/main/java/storage/DataFileReader.java:352]: Perhaps you could include the required params and exceptions thrown?

File: src/main/java/storage/DataFileWriter.java [hunk 3]
```diff
@@ -207,4 +215,28 @@ private void writeToFile(Path fileDirectory, Document xmlFileContents)
...
 
+    /**
+     * Gets the current settings and parses them into strings.
+     */
+    private void saveSettings() {
+        boolean isDisplayingHelpMessages = settings.getDisplayingHelpMessages();
+        int colorOption = settings.getColorOption();
+        StringBuilder saveData = new StringBuilder();
+        saveData.append("COLOR ").append(colorOption).append("\n");
+        saveData.append("HELPMESSAGE ").append(isDisplayingHelpMessages).append("\n");
```
Comment (theopin) [inline src/main/java/storage/DataFileWriter.java:226]: may wanna seperate each .append() to a new line for readability

**Review by theopin — APPROVED** [review]
Looks good so far! We can refine it once my pr on error handling is done

File: src/main/java/storage/DataFileReader.java (hunk 2 above)
Comment (adhy-p) [reply src/main/java/storage/DataFileReader.java:352]: ah yikes forgot about that HAHAH. thanks

File: src/main/java/storage/DataFileWriter.java (hunk 3 above)
Comment (adhy-p) [reply src/main/java/storage/DataFileWriter.java:226]: good point! will do

**Pull request merged**